rt/test/TESTOPR2.MLC

```diff
@@ -0,0 +1,1443 @@
+*----------------------------------------------------------------------
+         MNOTE '********** End   X2D tests **********'
+*======================================================================
+.SKPX2Y ANOP
+***********************************************************************
+*        MBOX  'Test DCLEN, DCVAL, DEQUOTE'
+         MBOX  'Test DCLEN'
+*AGO .SKPDC
+*======================================================================
+         MNOTE '********** Begin DCLEN tests **********'
+*----------------------------------------------------------------------
+*        HLASM LR examples
+
+&DCLEN   SETA  DCLEN('')                 null string
+         MNOTE 'DCLEN('''')=&DCLEN'
+         AIF   (&DCLEN NE 0).DCLERR
+&DCLEN   SETA  DCLEN('''')               single apostrophe
+         MNOTE 'DCLEN('''''''')=&DCLEN'
+         AIF   (&DCLEN NE 1).DCLERR
+&DCLEN   SETA  DCLEN('''''')             two apostrophes
+         MNOTE 'DCLEN('''''''''''')=&DCLEN'
+         AIF   (&DCLEN NE 1).DCLERR
+&DCLEN   SETA  DCLEN('&&')               two ampersands
+         MNOTE 'DCLEN(''&&&&'')=&DCLEN'
+         AIF   (&DCLEN NE 1).DCLERR
+&DCLEN   SETA  DCLEN('a''''b')           string is "a'b"
+         MNOTE 'DCLEN(''a''''''''b'')=&DCLEN'
+         AIF   (&DCLEN NE 3).DCLERR
+&DCLEN   SETA  DCLEN('a''''b&&c')        string is "a'b&c""
+         MNOTE 'DCLEN(''a''''''''b&&&&c'')=&DCLEN'
+         AIF   (&DCLEN NE 5).DCLERR
+&DCLEN   SETA  DCLEN('&&&&'.'''''''')    string is "&&''"
+         MNOTE 'DCLEN(''&&&&&&&&''.'''''''''''''''')=&DCLEN'
+         AIF   (&DCLEN NE 4).DCLERR
+*----------------------------------------------------------------------
+*        z390 tests
+
+&DCLSTR  SETC  '''''''''&&&&&&&&1'''''''''  len 17; see MNOTE below
+&K       SETA  K'&DCLSTR 
+&DDCLSTR SETC  DOUBLE('&DCLSTR')            double for MNOTE and DC
+*
+         MNOTE 'K''&&DCLSTR=&K  &&DCLSTR = >&DDCLSTR<'
+***********************************************************************
+*        On z390 the preceding MNOTE displays
+*
+*  MNOTE 'K'&DCLSTR=17  &DCLSTR = >''''''''&&&&&&&&&&&&&&&&1''''''''<'
+*
+*        Using HLASM the MNOTE displays
+*
+* +K'&DCLSTR=17  &DCLSTR = >''''&&&&&&&&1''''<           
+*
+*        Looks like mz390 just shows the MNOTE with the message
+*        text doubled, not the actual output of the MNOTE.
+*
+*        The following DC statement is the same for both z390 & HLASM.
+***********************************************************************
+         PUSH  PRINT
+         PRINT DATA
+         DC    FL1'&K',C'>&DDCLSTR<' 
+         POP   PRINT
+*
+&DCLEN   SETA  DCLEN('&DCLSTR') 
+         MNOTE 'DCLEN(''&&DCLSTR'')=&DCLEN' 
+         AIF   (&DCLEN NE 9).DCLERR
+*----------------------------------------------------------------------
+         MNOTE '********** End   DCLEN tests **********'
+*======================================================================
+         AGO   .SKPDCVO    DCVAL tests now done in rt/mlc/TOPR2.MLC
```

Comment:
if the tests have been moved, then it is (obviously) okay to document that here, but shouldn't the code be removed?

Reply:
I left it in so that one could compare "new" TOPR2.MLC version to "old" TESTOPR2.MLC version. Happy to delete it now if you think it is okay to do so.

Reply:
I think it adds more confusion than clarification. If anyone wants to go time travelling, the git repo supports that. No need to keep old stuff around.